pkg/config/config.go

```diff
@@ -187,36 +186,28 @@ func (c *ConfigImpl) GetCurrentContext() *AuthContext {
 // NewClient creates a new codefresh client for the current context or for
 // override context (if specified with --auth-context).
 func (c *ConfigImpl) NewClient() codefresh.Codefresh {
-	return c.clientForContext(c.GetCurrentContext())
+	client, err := c.clientForContext(c.GetCurrentContext())
+	if err != nil {
+		panic(err)
+	}
+
+	return client
 }
 
-func (c *ConfigImpl) NewAdHocClient(ctx context.Context, url, token string) (codefresh.Codefresh, error) {
+func (c *ConfigImpl) NewAdHocClient(ctx context.Context, url, token, caCert string) (codefresh.Codefresh, error) {
 	if url == "" {
 		url = store.Get().DefaultAPI
 	}
 
 	authCtx := &AuthContext{
-		Name:  "ad-hoc",
-		URL:   url,
-		Token: token,
-		Type:  "APIKey",
-		Beta:  false,
-	}
-
-	// validate new context
-	client := c.clientForContext(authCtx)
-	_, err := client.Users().GetCurrent(ctx)
-	if err != nil {
-		if url == store.Get().DefaultAPI {
-			err = fmt.Errorf("failed to create codefresh client with token \"%s\": %w", token, err)
-		} else {
-			err = fmt.Errorf("failed to create codefresh client with url \"%s\" and token \"%s\": %w", url, token, err)
-		}
```

Comment:
removed check on client creation, because now we can create a client with a runtimeToken (which fails on `GetCurrent` calls)